server/datastore/mysql/software_installers.go

```diff
@@ -2,14 +2,76 @@ package mysql
 
 import (
 	"context"
-	"database/sql"
+ 	"database/sql"
 
 	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
 	"github.com/fleetdm/fleet/v4/server/fleet"
 	"github.com/google/uuid"
 	"github.com/jmoiron/sqlx"
 )
 
+func (ds *Datastore) ListPendingSoftwareInstallDetails(ctx context.Context, hostID uint) ([]*fleet.SoftwareInstallDetails, error) {
+	const stmt = `
+  SELECT
+    hsi.host_id AS host_id,
+    hsi.execution_id AS execution_id,
+    hsi.software_installer_id AS installer_id,
+    si.pre_install_query AS pre_install_condition,
+    inst.contents AS install_script,
+    pinst.contents AS post_install_script
+  FROM
+    host_software_installs hsi
+  INNER JOIN
+    software_installers si
+    ON hsi.software_installer_id = si.id
+  LEFT JOIN
+    script_contents inst
+    ON inst.id = si.install_script_content_id
+  LEFT JOIN
+    script_contents pinst
+    ON pinst.id = si.post_install_script_content_id
+  WHERE
+    hsi.host_id = ?
+  AND
+    hsi.install_script_exit_code IS NULL
```

Comment:
I think you _also_ want to check that `hsi.pre_install_query_output` is `NULL`, because if the pre-condition fails, the `install_script_exit_code` will stay `NULL` but the software installation is not pending anymore. The pre-condition is optional, so you need to check both.